examples/common/python/shared_kv/remote_lmdb/lmdb_listener.py

```diff
@@ -27,92 +28,105 @@
 TCFHOME = os.environ.get("TCF_HOME", "../../../../")
 
 # -----------------------------------------------------------------
+
+
 def local_main(config):
 
     root = LMDBRequestHandler(config)
     site = server.Site(root)
-    reactor.listenTCP(int(bind_port), site)
-
+    reactor.listenTCP(bind_port, site)
     logger.info('LMDB Listener started on port %s', bind_port)
 
-    try :
+    try:
         reactor.run()
     except ReactorNotRunning:
         logger.warn('shutdown')
-    except :
+    except:
         logger.warn('shutdown')
 
     exit(0)
 
 # -----------------------------------------------------------------
-def parse_command_line(config, args) :
+
+
+def parse_command_line(config, args):
 
     global bind_port
 
+    if config.get("KvStorage") is None or config["KvStorage"].get("remote_url") is None:
```

Comment:
you meant "and" instead of "or" ?

if (config.get("KvStorage") is None) and (config["KvStorage"].get("remote_url") is None):

Reply:
`or` not `and`：either `KvStorage` or `KvStorage.remote_url` shouldn't employ the remote LMDB @manojgop

Reply:
either StoragePath or remote_url should be present. If both are None then its an error condition

if (config.get("KvStorage").get("StoragePath) is None) and (config["KvStorage"].get("remote_url") is None):

Reply:
Yes, it should error out in case of both being None.

For `if (config.get("KvStorage").get("StoragePath") is None) and (config["KvStorage"].get("remote_url") is None)`, 
the chaining call `config.get("KvStorage").get("StoragePath")` would throw in case of `config.get('KvStorage')` being `None`~

For ` if config.get("KvStorage") is None or config["KvStorage"].get("remote_url") is None`,
if `config.get("KvStorage") is None` is `true`, we can early abort since `config["KvStorage"].get("remote_url")` mustn't exist then. Otherwise (i.e., `config["KvStorage"]` isn't `None`), we still need to ensure that `remote_url` goes under it. Only then should we employ the remote LMDB safely without crashing the program ~ @manojgop

Reply:
What I meant is if both StoragePath and remote_url is not set i.e if both are None then it's an error condition.  If "KvStorage" is none then it's an error condition.

Reply:
config loader should have the ability to ensure that the given configuration is verified and rightful before passing to other modules, instead of checking parameter everywhere. Maybe we can make a TODO in the `config/config.py`.

Reply:
I got it. 

However, if the `and` check is taken,  we would miss out the error case with `KvStorage.StoragePath` set and `KvStorage.remote_url` unset, which should quit the listener due to no suitable url to serve on.

But `lmdb_listener` should only care whether `KvStorage.remote_url` is set, and leave the checking against `KvStorage.StoragePath` to the `LMDBRequestHandler`.

Reply:
ok.